Lib/html/parser.py

```diff
@@ -57,6 +58,26 @@
 # </ and the tag name, so maybe this should be fixed
 endtagfind = re.compile(r'</\s*([a-zA-Z][-.a-zA-Z0-9:_]*)\s*>')
 
+# Character reference processing logic specific to attribute values
+# See: https://html.spec.whatwg.org/multipage/parsing.html#named-character-reference-state
+attr_charref = re.compile(r'&(#[0-9]+|#[xX][0-9a-fA-F]+|[a-zA-Z][a-zA-Z0-9]*)[;=]?')
```

Comment:
This partially duplicates an existing Regex, but I was not able to reuse the existing one for this purpose.

Reply:
Since the issue only seems to affect named character references, is there a reason to include numeric charrefs too in this regex?

Reply:
Yes, the new `_unescape_attrvalue` is effectively a wrapper for `html.escape` that only delegates to `html.escape` if the attribute specific conditions are met. Since we still want to escape numeric and hex char refs in attributes, we need to include them in the regex.